### PR TITLE
Attachments in compose and reply

### PR DESCRIPTION
**Pull request by KamilaHasanbega**
Fixed the page for replying with attachments, and added tests for the new onFileUpload function. Fixed other tests to include attachments.

### CODE REVIEW

File: chrome/pages/threads/threads-controller.js [hunk 1]
```diff
@@ -457,9 +457,12 @@ ThreadsCtrl.prototype.isInviteInProgress = function() {
 
 
 /**
+<<<<<<< HEAD
```
Comment (koto) [inline chrome/pages/threads/threads-controller.js]: This and the above are file conflict markers, you should resolve the conflicts by deleting the lines in those files that shouldn't be there (`<<<<`, `====` and `>>>` lines should not be committed).

File: chrome/pages/messages/messages.css [hunk 2]
```diff
@@ -1,10 +1,11 @@
 .email-detail-entry {
   border-bottom: 1px solid rgba(0,0,0,0.04);
   margin-bottom: 1em;
+  text-align: left !important;
 }
 
-.email-detail-from {
-
+.email-detail-from { 
+  text-align: left !important;
```
Comment (koto) [inline chrome/pages/messages/messages.css]: text-align: left shouldn't be needed as the text-align: center was fixed in e19e0e39deca66683b8a066fcc91e272cdf60a8e.

File: chrome/pages/reset/reset.html [hunk 3]
```diff
@@ -17,3 +17,4 @@ <h1 class="wizard-title text-center"><b>{{'resetTitle' | translate}}</b></h1>
   </div>
 </div>
 </div>
```
Comment (koto) [inline chrome/pages/reset/reset.html]: There too many `</div>`s (in fact, two of them should be removed, not just one)

File: chrome/pages/messages/messages-controller.js [hunk 4]
```diff
@@ -95,6 +96,36 @@ MessagesCtrl.prototype.cancelReply = function(opt_event) {
   this.reply['baseTitle'] = 'reply';
   this.reply['content'] = null;
   this.reply['showText'] = false;
+  this.reply['attachments'] = [];
+};
+
+
+/**
```
Comment (koto) [inline chrome/pages/messages/messages-controller.js]: The function description is missing, same below.

File: chrome/pages/messages/messages-controller.js [hunk 5]
```diff
@@ -95,6 +96,36 @@ MessagesCtrl.prototype.cancelReply = function(opt_event) {
...
+ * @param {string} name of the file
```
Comment (koto) [inline chrome/pages/messages/messages-controller.js]: 'name of the file' => "The name of the file.", same for other parameters.

File: chrome/pages/messages/messages-controller.js [hunk 6]
```diff
@@ -66,12 +66,13 @@ e2email.pages.messages.MessagesCtrl = function(
   /**
    * Contains the state related to any replies by the user
    * for this thread.
-   * @type {!{baseTitle: string, showText: boolean, content: ?string}}
+   * @type {!{baseTitle: string, showText: boolean, content: ?string, attachments: Array<Object>}}
```
Comment (koto) [inline chrome/pages/messages/messages-controller.js]: Could we make a `@typedef`  of the attachments and make `attachments` and `Array<...Attachment>' ? The mail model file would be a good fit for this new typedef.

File: chrome/models/mail/mail-model.js [hunk 7]
```diff
@@ -18,12 +18,29 @@
...
 
+goog.provide('e2email.models.mail.Attachments');
 goog.provide('e2email.models.mail.Mail');
 goog.provide('e2email.models.mail.Mailbox');
 goog.provide('e2email.models.mail.Thread');
 
 
 /**
+ * This is the model for the attachments support.
+ * The filename is the attachment's name, size is it's size in Bytes,
```
Comment (koto) [inline chrome/models/mail/mail-model.js]: some typos: Bytes -> bytes and the ocntent.

File: chrome/pages/messages/messages.html [hunk 8]
```diff
@@ -38,22 +38,27 @@ <h4 class="col-xs-12 page-header email-subject selectable">{{messagesCtrl.thread
...
       </div>
     </li>
   </ul>
 
   <form class="email-detail-reply-form fade-show" ng-show="messagesCtrl.thread.mails[messagesCtrl.thread.mails.length-1].mimeContent && !messagesCtrl.thread.mails[messagesCtrl.thread.mails.length-1].hasErrors" ng-submit="messagesCtrl.onReply()">
-    <textarea ng-show="messagesCtrl.reply.showText" class="form-control fade-show" placeholder="{{'typeReplyPlaceholder' | translate}}" ng-model="messagesCtrl.reply.content" rows="10"></textarea>
+    <div id="textareaSpace"  ng-show="messagesCtrl.reply.showText">
+      <textarea id="textareaElement" class="form-control fade-show" placeholder="{{'typeReplyPlaceholder' | translate}}" ng-model="messagesCtrl.reply.content" rows="10"></textarea>
+    <div id="attachmentRepeat">
+        <div ng-repeat="x in messagesCtrl.reply.attachments" class="single-attachment"><span style="margin: 0px; padding:0px;overflow-y: hidden; width:60%;">{{ x.filename | limitTo: 20 }}{{x.filename.length > 20 ? '...' : ''}} </span> <span style="width:30%; overflow-y:hidden;"> {{' (' + x.size/1000 +
```
Comment (koto) [inline chrome/pages/messages/messages.html]: Move the inline style rules to some CSS class.

File: chrome/pages/messages/messages.css [hunk 9]
```diff
@@ -25,6 +25,7 @@
 .email-subject {
   font-family: Roboto;
   color: #505050;
+  text-align: left !important;
```
Comment (koto) [inline chrome/pages/messages/messages.css]: text-align: left is probably not needed here and in .email-text-content

File: chrome/models/mail/mail-model.js (hunk 7 above)
Comment (koto) [inline chrome/models/mail/mail-model.js]: it's size => its size

File: chrome/models/mail/mail-model.js [hunk 10]
```diff
@@ -18,12 +18,29 @@
...
+ * with it's ocntent, type is the atttachment's type.
+ * @typedef {{
+ * filename: string,
+ * type: string,
+ * encoding: string,
+ * content: string,
+ * size: number
+ * }}
+ */
+e2email.models.mail.Attachments;
```
Comment (koto) [inline chrome/models/mail/mail-model.js]: This should be singular, as it defines a single attachment.

File: chrome/pages/messages/messages.css [hunk 11]
```diff
@@ -3,7 +3,7 @@
   margin-bottom: 1em;
 }
 
-.email-detail-from {
+.email-detail-from {
```
Comment (koto) [inline chrome/pages/messages/messages.css]: That's a selector with no definitions, please remove.

File: chrome/pages/messages/messages.html [hunk 12]
```diff
@@ -38,22 +38,31 @@ <h4 class="col-xs-12 page-header email-subject selectable">{{messagesCtrl.thread
       </div>
       <div class="row email-mime-entry" ng-repeat="mime in mail.mimeContent" ng-show="mail.mimeContent">
         <img class="fade-show email-image-content" ng-src="{{mime.content}}" ng-if="mime.type=='image'">
-        <div class="email-text-content col-xs-12" ng-if="mime.type=='text'">{{mime.content}}
-        </div>
-        <div class="col-xs-12" ng-if="mime.type=='unsupported'">
-          <div class="col-xs-12 email-unsupported-content text-warning">{{mime.content}}</div>
-        </div>
+        <div class="email-text-content col-xs-12" ng-if="mime.type=='text'">{{mime.content}}</div>
+        <div class="col-xs-12 single-attachment-received" ng-if="mime.url"> <a href="{{mime.url}}" download>{{mime.filename}}</div>
```
Comment (koto) [inline chrome/pages/messages/messages.html]: The download attribute can have a value https://developers.google.com/web/updates/2011/08/Downloading-resources-in-HTML5-a-download : Please use the download="{{mime.filename}}"

**Comment by koto** [comment]
LGTM, safe to merge.

**Comment by kbsriram** [comment]
Rebased via 86185ae

**Pull request closed without merging**